src/main/java/uschi2000/benchmark/GrpcServer.java

```diff
@@ -72,7 +79,9 @@ public void stop() {
 
         @Override
         public void query(BenchmarkRequest req, StreamObserver<BenchmarkReply> responseObserver) {
+            String authHeader = AuthHeaders.BEARER_TOKEN.get(Context.current());
```

Comment:
we could also have every gRPC request we generate include the auth header - a bit more testable...

Reply:
this would have made it very easy it very easy to implement your ledger based auth thing btw...

Reply:
i think the auth header should be explicit, not sure how to best do that...

Reply:
we can pass around as a request param probably - just experiment. I think i agree